x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/RealmsTests.java

```diff
@@ -223,6 +262,10 @@ public void testUnlicensedWithOnlyCustomRealms() throws Exception {
         Environment env = TestEnvironment.newEnvironment(settings);
         Realms realms = new Realms(settings, env, factories, licenseState, threadContext, reservedRealm);
 
+        allowAllRealms();
+        verify(licenseState, atLeastOnce()).checkFeature(Feature.SECURITY);
+        verify(licenseState, atLeastOnce()).checkFeature(Feature.SECURITY_CUSTOM_REALM);
```

Comment:
Took me a while to understand the purpose of these two checks: IIUC, they are the alternative to ensure the licenseState listener work. Since the listener is a protected method and the its downstream method `isRealmTypeAvailable` is static, so the interactions have to be checked like this. I think it could use some comments for the underlying logic to help future readers (future me included).

Reply:
I'll add comments, but the explanation isn't what you've described.
The _purpose_ of this PR is to add these calls. This is how we track feature usage. So, the test verifies that they've been called.